qiita_db/data.py

```diff
@@ -1257,6 +1257,60 @@ def create(cls, processed_params_table, processed_params_id, filepaths,
         pd.add_filepaths(filepaths, conn_handler)
         return cls(pd_id)
 
+    @classmethod
+    def delete(cls, processed_data_id):
+        """Removes the processed data with id processed_data_id
+
+        Parameters
+        ----------
+        processed_data_id : int
+            The processed data id
+
+        Raises
+        ------
+        QiitaDBUnknownIDError
+            If the processed data id doesn't exist
+        QiitaDBStatusError
+            If the processed data status is not sandbox
+        """
+        if cls(processed_data_id).status != 'sandbox':
+            raise QiitaDBStatusError(
+                "Illegal operation on public processed data")
+
+        conn_handler = SQLConnectionHandler()
+
+        analysis_ids = [str(_id[0]) for _id in conn_handler.execute_fetchall(
+            "SELECT DISTINCT analysis_id FROM qiita.analysis_sample WHERE "
+            "processed_data_id = {0} ORDER BY "
+            "analysis_id".format(processed_data_id))]
+
+        if analysis_ids:
+            raise QiitaDBError(
+                "Processed data %d is linked to (meta)analyses: %s" %
+                (processed_data_id, ", ".join(analysis_ids)))
+
+        # delete
+        queue = "delete_processed_data_%d" % processed_data_id
+        conn_handler.create_queue(queue)
+
+        sql = ("DELETE FROM qiita.preprocessed_processed_data WHERE "
+               "processed_data_id = {0}".format(processed_data_id))
+        conn_handler.add_to_queue(queue, sql)
+
+        sql = ("DELETE FROM qiita.processed_filepath WHERE "
+               "processed_data_id = {0}".format(processed_data_id))
+        conn_handler.add_to_queue(queue, sql)
+
+        sql = ("DELETE FROM qiita.study_processed_data WHERE "
+               "processed_data_id = {0}".format(processed_data_id))
+        conn_handler.add_to_queue(queue, sql)
+
+        sql = ("DELETE FROM qiita.processed_data WHERE "
+               "processed_data_id = {0}".format(processed_data_id))
+        conn_handler.add_to_queue(queue, sql)
+
+        conn_handler.execute_queue(queue)
+
```

Comment:
For both comments: What we are currently doing is to leave the filepath entry (it doesn't affect anything) as deleting files could be pretty expensive. Additionally, the plan is to drop one of the already created functions once we have this in: https://github.com/biocore/qiita/issues/814